### ISSUE TITLE
Cannot read properties of undefined (reading '0')

### ISSUE THREAD

**Issue by quotschmacher** [bryanbcook/azdevops-testplan-extension#16]
**Description**

The XML xUnit file cannot be processed.

```yaml
steps:
- task: bcook.azdevops-testplan-extension.publishtestplanresults.PublishTestPlanResults@0
  displayName: 'Publish Test Results'
  inputs:
    accessToken: '$(System.AccessToken)'
    testPlan: 'UI Test mit Testcafe'
    testResultFormat: xUnit
    testResultFiles: testcafe/report.xml
    testCaseMatchStrategy: name
```

**Environment Details**

- Pool: Azure Pipelines
- Image: ubuntu-22.04
- Queued: Today at 11:24 
- Agent: Hosted Agent
- - pipeline-format: Designer-based (Classic)

***The xUnit file***

- report file generated with the Testcafe xUnit reporter

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuite name="TestCafe Tests: Firefox 122.0 / Ubuntu 22.04" tests="2" failures="0" skipped="0" errors="0" time="211.682" timestamp="Tue, 13 Feb 2024 10:28:58 GMT" >
  <testcase classname="Neuland Tests" file="/home/vsts/work/1/s/testcafe/tests/testNld.js" name="LogIn" time="15.513">
  </testcase>
  <testcase classname="Neuland Tests" file="/home/vsts/work/1/s/testcafe/tests/testNld.js" name="Auftrag erfassen" time="193.695">
  </testcase>
</testsuite>
```

**Pipeline Output**

_provide output from the build with diagnostics enabled (system.debug variable set to 'true'). Mask secrets or sensitive information with xxx_:

```shell
##[debug]Evaluating condition for step: 'Publish Test Results'
##[debug]Evaluating: succeeded()
##[debug]Evaluating succeeded:
##[debug]=> True
##[debug]Result: True
Starting: Publish Test Results
==============================================================================
Task         : Publish Test Plan Results
Description  : Publishes cross-platform test results (xUnit, JUnit) to an Azure DevOps Test Plan
Version      : 0.1.10
Author       : Bryan Cook
Help         : 
==============================================================================
##[debug]Using node path: /home/vsts/agents/3.232.3/externals/node16/bin/node
[...]
Available Test Points: 2
##[debug]reading TestFrameworkParameters from task inputs.
##[debug]testResultFormat=xUnit
##[debug]testResultDirectory=undefined
##[debug]testResultFiles=/home/vsts/work/1/s/testcafe/report.xml
##[debug]check path : /home/vsts/work/1/s/testcafe/report.xml
##[debug]converting test framework results into unified format
##[debug]task result: Failed
##[error]Cannot read properties of undefined (reading '0')
##[debug]Processed: ##vso[task.issue type=error;]Cannot read properties of undefined (reading '0')
##[debug]Processed: ##vso[task.complete result=Failed;]Cannot read properties of undefined (reading '0')

```

**Comment by quotschmacher** [comment]
Just looked up some guideline about xUnit and found out that my xml file looks more like jUnit. But setting the task up to interprete a jUnit file gives the same error.

**Comment by bryanbcook** [comment]
The 3rd party library I'm using to read through the test results doesn't like the file that you're providing.

The xml you supplied is definitely not xUnit. Take a look at some of the sample test data in the source code: PublishTestPlanResultsV1/test/data

Can you attach your xml file? A simple unit test should help us identify what aspect of the xml file has the issue. I suspect that your xml file is missing attributes that the parser is expecting.

**Comment by quotschmacher** [comment]
thanks for your research and effort!

your first reference example states the following at https://github.com/testmoapp/junitxml?tab=readme-ov-file#complete-junit-xml-example :

> Usually the root element of a JUnit XML file. Some tools leave out 
the <testsuites> element if there is only a single top-level <testsuite> element (which
is then used as the root element).

azure dev ops is for example such a tool that excepts it without a root-element. 

i will see what i can do but as this is not your fault i will close this ticket. thanks a lot - i will go on trying to get your extension running for me ;-)

**Comment by bryanbcook** [comment]
I wouldn't worry about creating a custom reporter. As per the thread above, I've submitted a PR to the test-parser library that this project uses. The PR has been merged, but I need the owner to publish the changes to the npm repository. The work item for that is here: https://github.com/test-results-reporter/parser/issues/46

As soon as the owner of that project creates a release with the changes, I can update publish a new version of the extension with the updated dependency..
